### PR TITLE
tests: Cross-node http poststart / prestop hooks

### PR DESCRIPTION
**Pull request by claudiubelu**
#### What type of PR is this?

/kind feature
/sig testing
/sig networking

#### What this PR does / why we need it:

In the case of http poststart / prestop hooks, ``kubelet`` itself is responsible for making the http call. These hooks may be able to execute without any issue on the same node, but that might not be true for cross-node. This commit adds tests that verifies cross-node hooks.

Note: exec poststart / prestop hooks are done by exec-ing into the containers, and cross-node pod-pod communcation should be fine.

#### Which issue(s) this PR fixes:

Related: #101062

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/common/node/lifecycle_hook.go [hunk 1]
```diff
@@ -127,6 +154,27 @@ var _ = SIGDescribe("Container Lifecycle Hook", func() {
...
+				},
+			}
+			podWithHook := getPodWithHook("pod-with-poststart-http-hook", imageutils.GetPauseImageName(), lifecycle)
+
+			// make sure we spawn the test pod on a different node than the webserver.
+			nodeSelection := e2epod.NodeSelection{}
+			e2epod.SetAntiAffinity(&nodeSelection, targetNode)
+			e2epod.SetNodeSelection(&podWithHook.Spec, nodeSelection)
+
 			testPodWithHook(podWithHook)
```
Comment (oomichi) [inline test/e2e/common/node/lifecycle_hook.go:178]: ditto

**Review by oomichi — COMMENTED** [review]
/cc @oomichi

**Comment by aojea** [comment]
/assign @thockin 
we've talked today in the sig-network meeting about this

**Review by thockin — COMMENTED** [review]
So if we say this is NOT a requirement, how do we change this test?

**Comment by aojea** [comment]
I think that at this point we should not test it, it will create confusion only

**Comment by thockin** [comment]
Agree

**Comment by aojea** [comment]
> I think that at this point we should not test it, it will create confusion only

>> Agree

/hold 
just to avoid we inadvertently merge this
sorry for the back and forth :sweat_smile:

**Pull request closed without merging**